tmt/base.py

```diff
@@ -2710,8 +2710,20 @@ def go(self) -> None:
         self.save()
 
         # Iterate over plans
+        crashed_plans: List[Tuple[Plan, Exception]] = []
+
         for plan in self.plans:
-            plan.go()
+            try:
+                plan.go()
+
+            except Exception as exc:
+                if self.opt('on-plan-error') == 'quit':
+                    raise exc
+
+                crashed_plans.append((plan, exc))
+
+        if crashed_plans:
+            raise crashed_plans[0][1]
```

Comment:
I'd prefer to list all crashed plans and the reasons. There could be several of them with different reasons (e.g. invalid image in one , incorrect require in another etc.)

Reply:
There's only a single exception that can be raised, and it can have a single "cause" only (unless Python 3.11 & https://peps.python.org/pep-0654/ becomes a daily routine). All exceptions are logged already, AFAICT, by `plan.go()`, before they bubble up to this point - none of them is lost. Only the first one is used as a "cause" of why the run failed to finish.

How about adding a dedicated exception for this situation, something like `RunFailedError` which would accept multiple "causes", and a special handling in `tmt.utils.render_exception()` to honor not just the default Python chaining via `__cause__`, but also iterate over `RunFailedError.causes`?

Reply:
Fixed in 99bea1a6